src/routes/StudentRequests/StudentRequests.tsx

```diff
@@ -0,0 +1,285 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { sortStudents } from 'utils/sortStudentRequests/sortStudents';
+import SectionHeader from 'components/SectionHeader/SectionHeader';
+import { InputField } from 'components/InputField/InputField';
+import { Button } from 'components/Button/Button';
+import { StudentRequest } from 'types/StudentRequest/StudentRequest';
+import StudentRequestTable, {
+  SortRule,
+  HEADERS,
+} from 'components/StudentRequestTable/StudentRequestTable';
+
+import { RootState } from 'store/store';
+import {
+  getStudentRequests,
+  respondOneStudentRequest,
+  respondManyStudentRequest,
+} from 'utils/db/db.utils';
+import { setLoading, removeLoading } from 'store/loading/loadingSlice';
+import { updateToast, TOAST_GENERAL_ERRORS } from 'store/toast/toastSlice';
+import { useSelector, useDispatch } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+
+import style from './StudentRequests.module.css';
+
+const ALL_GROUPS: string = 'Todos los grupos';
+
+type Props = {
+  initialClass?: string;
+};
+
+export default function StudentRequests({ initialClass }: Props) {
+  const autoComplete = useRef<HTMLInputElement>(null);
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const user = useSelector((state: RootState) => state.user.currentUser);
+
+  // Complete array of Student Requests
+  const [data, setData] = useState<StudentRequest[]>([]);
+  const [updateFetch, setUpdateFetch] = useState<number>(0);
+  const [isListOpen, setIsListOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [filterValue, setFilterValue] = useState<string>(
+    initialClass as string
+  );
+  const [selectedRows, setSelectedRows] = useState<string[]>([]);
+  const [sortRule, setSortRule] = useState<SortRule>({
+    element: HEADERS[0],
+    value: 'Up',
+  });
+
+  // Update information for input
+  useEffect(() => {
+    if (autoComplete.current) autoComplete.current.value = filterValue;
+  }, [filterValue]);
+
+  // Get information from db on requests
+  useEffect(() => {
+    if (user?.role !== 'teacher') {
```

Comment:
Maybe we should create a Route Wrapper so that it checks whether the user role is correct before rendering the page